lib/specialist_tag_finder.rb

```diff
@@ -1,32 +1,31 @@
 class SpecialistTagFinder
 
-  def initialize(document)
-    @document = document
+  def initialize(edition)
+    @edition = edition
   end
 
-  def primary_sector_tag
-    primary_subsector_tag.parent if primary_subsector_tag
+  def topics
+    presented_edition = PublishingApiPresenters::Edition.new(@edition)
+    edition_path = presented_edition.base_path
+    content_item = Whitehall.content_store.content_item(edition_path)
+    return [] unless content_item
+    Array(content_item.links["topics"])
   end
 
-  def primary_subsector_tag
-    if primary_tag_slug = @document.primary_specialist_sector_tag
-      specialist_sector_tags.find {|t| t.slug == primary_tag_slug }
-    end
+  def grandparent_topic
+    presented_edition = PublishingApiPresenters::Edition.new(@edition)
+    edition_path = presented_edition.base_path
+    edition_content_item = Whitehall.content_store.content_item(edition_path)
+    return unless edition_content_item
+    parents = Array(edition_content_item.links["parent"])
+    return unless parents.any?
+
+    # FIXME: We now need to fetch the parent topic from the content store to
```

Comment:
You may be able to retrieve them both at once from content store, making it do the resolution. See https://github.com/alphagov/content-store/pull/203